File: mobile/domain/src/main/java/com/example/domain/budget/model/Budget.kt
package com.example.domain.budget.model

data class Budget(
    val id: Int,
    val month: Int,
    val year: Int,
    val monthlyLimit: Double,
    val dailyLimit: Double,
    val monthlySpent: Double,
    val dailySpent: Double,
    val lastUpdatedDate: String,
    val createdAt: String
) {
    val monthlyRemaining: Double
        get() = monthlyLimit - monthlySpent

    val dailyRemaining: Double
        get() = dailyLimit - dailySpent

    val monthlyProgress: Float
        get() = if (monthlyLimit > 0) (monthlySpent / monthlyLimit).toFloat() else 0f

    val dailyProgress: Float
        get() = if (dailyLimit > 0) (dailySpent / dailyLimit).toFloat() else 0f

    val isMonthlyExceeded: Boolean
        get() = monthlySpent > monthlyLimit

    val isDailyExceeded: Boolean
        get() = dailySpent > dailyLimit
}

File: backend/app/entities/budget.py
from sqlalchemy import Column, ForeignKey, Integer, DateTime, DECIMAL, Date
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base
from datetime import date

class Budget(Base):
    __tablename__ = "budgets"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False)

    monthly_limit = Column(DECIMAL(10, 2), default=0.00)
    daily_limit = Column(DECIMAL(10, 2), default=0.00)

    monthly_spent = Column(DECIMAL(10, 2), default=0.00)
    daily_spent = Column(DECIMAL(10, 2), default=0.00)

    month = Column(Integer, nullable=False)
    year = Column(Integer, nullable=False)

    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    last_updated_date = Column(Date, default=date.today)

    user = relationship("User", back_populates="budgets")